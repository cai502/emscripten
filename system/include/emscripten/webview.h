extern int iframe_create(void);
extern void iframe_destroy(int id);
extern void iframe_attach(int id);
extern void iframe_detach(int id);
extern void iframe_setFrame(int id, int left, int top, int width, int height);
extern void iframe_loadUrl(int id, const char* url);
extern void iframe_loadString(int id, const char* str);
extern void iframe_stopLoading(int id);
extern void iframe_reload(int id);
extern void iframe_goBack(int id);
extern void iframe_goForward(int id);
