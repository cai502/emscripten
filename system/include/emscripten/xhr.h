#include <dispatch/queue.h>

#if defined(__cplusplus)
#define XHR_EXTERN extern "C"
#else
#define XHR_EXTERN extern
#endif

XHR_EXTERN int _xhr_create(void);
XHR_EXTERN void _xhr_open(int xhr, const char *method, const char *url, int async, const char *user, const char *password);
XHR_EXTERN void _xhr_clean(int xhr);
XHR_EXTERN void _xhr_abort(int xhr);
XHR_EXTERN void _xhr_set_onload(int xhr, dispatch_queue_t queue, void *ctx, void func(void*));
XHR_EXTERN void _xhr_set_onerror(int xhr, dispatch_queue_t queue, void *ctx, void func(void*));
XHR_EXTERN void _xhr_set_request_header(int xhr, const char *key, const char *value);
XHR_EXTERN void _xhr_set_with_credentials(int xhr, int with_credentials);
XHR_EXTERN void _xhr_set_timeout(int xhr, int timeout);
XHR_EXTERN void _xhr_send(int xhr, const void *data, int length);
XHR_EXTERN int _xhr_get_ready_state(int xhr);
XHR_EXTERN int _xhr_get_status(int xhr);
XHR_EXTERN int _xhr_get_status_text(int xhr, void **text); // return length, text needs to be freed by caller
XHR_EXTERN int _xhr_get_response_text(int xhr, void **data); // return length, data needs to be freed by caller
XHR_EXTERN int _xhr_get_all_response_headers(int xhr, void **data); // return length, data needs to be freed by caller

#undef XHR_EXTERN
