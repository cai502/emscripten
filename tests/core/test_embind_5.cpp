#include <emscripten.h>
#include <emscripten/bind.h>

using namespace emscripten;
class MyFoo {
public:
    MyFoo() {
        EM_ASM({print("constructing my foo");});
    }
    void doit() {
        EM_ASM({print("doing it");});
    }
};
EMSCRIPTEN_BINDINGS(my_module) {
    class_<MyFoo>("MyFoo")
      .constructor<>()
      .function("doit", &MyFoo::doit)
      ;
}
int main(int argc, char **argv) {
    EM_ASM(
      try {
        var foo = new Module.MyFoo();
        foo.doit();
      } catch(e) {
        Module.print(e);
      }
    );
    return 0;
}
