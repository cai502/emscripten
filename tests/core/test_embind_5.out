constructing my foo
doing it